src/app/icd/client/DefaultICDClientStorage.cpp

```diff
@@ -161,7 +161,13 @@ void DefaultICDClientStorage::ICDClientInfoIteratorImpl::Release()
 CHIP_ERROR DefaultICDClientStorage::Init(PersistentStorageDelegate * clientInfoStore, Crypto::SymmetricKeystore * keyStore)
 {
     VerifyOrReturnError(clientInfoStore != nullptr && keyStore != nullptr, CHIP_ERROR_INVALID_ARGUMENT);
+    if (mpClientInfoStore != nullptr && mpKeyStore != nullptr && mpClientInfoStore == clientInfoStore && mpKeyStore == keyStore)
+    {
+        ChipLogProgress(ICD, "DefaultICDClientStorage has already been initialized");
```

Comment:
Though realistically: the right response to multiple calls on `Init()` on an object is to error out of all but the first, because clearly the API consumer is just completely confused.  So actually, I don't understand why we would want to make this change.

Reply:
yes, I agree I think we may introduce a function call IsInitialized() and a state(Uninitialized/Initialized)for ICDClientStorage, it seems to be more natural to me
just update the code to reflect this proposal

Reply:
Just update code to shutdown icd client storage when destroying android controller, thanks